commit 10444f68fb25bc0eebd4f161a58c494f68cce32f
Author: Alvin Wong <alvinhochun@gmail.com>
Date:   Sun Oct 8 02:04:26 2017 +0800

    Revert "Windows QPA: Hardcode a limit for the default point size"
    
    This reverts commit a72513cab7cdfac638ef572838277aa062f1d296.

diff --git a/qtbase/src/platformsupport/fontdatabases/windows/qwindowsfontdatabase.cpp b/qtbase/src/platformsupport/fontdatabases/windows/qwindowsfontdatabase.cpp
index 58b700b93f..d3e4daa341 100644
--- a/qtbase/src/platformsupport/fontdatabases/windows/qwindowsfontdatabase.cpp
+++ b/qtbase/src/platformsupport/fontdatabases/windows/qwindowsfontdatabase.cpp
@@ -1621,7 +1621,6 @@ void QWindowsFontDatabase::refUniqueFont(const QString &uniqueFont)
         m_uniqueFontData[uniqueFont].refCount.ref();
 }
 
-// ### fixme Qt 6 (QTBUG-58610): See comment at QWindowsFontDatabase::systemDefaultFont()
 HFONT QWindowsFontDatabase::systemFont()
 {
     static const HFONT stock_sysfont = (HFONT)GetStockObject(DEFAULT_GUI_FONT);
@@ -1962,31 +1961,12 @@ QFontEngine *QWindowsFontDatabase::createEngine(const QFontDef &request, const Q
 
 QFont QWindowsFontDatabase::systemDefaultFont()
 {
-#if QT_VERSION >= 0x060000
-    // Qt 6: Obtain default GUI font (typically "Segoe UI, 9pt", see QTBUG-58610)
-    NONCLIENTMETRICS ncm;
-    ncm.cbSize = FIELD_OFFSET(NONCLIENTMETRICS, lfMessageFont) + sizeof(LOGFONT);
-    SystemParametersInfo(SPI_GETNONCLIENTMETRICS, ncm.cbSize , &ncm, 0);
-    const QFont systemFont = QWindowsFontDatabase::LOGFONT_to_QFont(ncm.lfMessageFont);
-#else
     LOGFONT lf;
     GetObject(QWindowsFontDatabase::systemFont(), sizeof(lf), &lf);
     QFont systemFont =  QWindowsFontDatabase::LOGFONT_to_QFont(lf);
     // "MS Shell Dlg 2" is the correct system font >= Win2k
     if (systemFont.family() == QLatin1String("MS Shell Dlg"))
         systemFont.setFamily(QStringLiteral("MS Shell Dlg 2"));
-    // Qt 5 by (Qt 4) legacy uses GetStockObject(DEFAULT_GUI_FONT) to
-    // obtain the default GUI font (typically "MS Shell Dlg 2, 8pt"). This has been
-    // long deprecated; the message font of the NONCLIENTMETRICS structure obtained by
-    // SystemParametersInfo(SPI_GETNONCLIENTMETRICS) should be used instead (see
-    // QWindowsTheme::refreshFonts(), typically "Segoe UI, 9pt"), which is larger.
-    // In single monitor setups, the point sizes revolve around 8 (depending on UI
-    // scale factor, but not proportional to it). However, in multi monitor setups,
-    // where the DPI of the primary monitor are smaller than those of the secondary,
-    // large bogus values are returned. Limit to 8.25 in that case.
-    if (GetSystemMetrics(SM_CMONITORS) > 1 && systemFont.pointSizeF() > 8.25)
-        systemFont.setPointSizeF(8.25);
-#endif // Qt 5
     qCDebug(lcQpaFonts) << __FUNCTION__ << systemFont;
     return systemFont;
 }
